libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/RestRequest.java

```diff
@@ -836,6 +833,91 @@ public static RestRequest getRequestForPrimingRecords(String apiVersion, String
 		return new RestRequest(RestMethod.GET, path.toString());
 	}
 
+	/**
+	 * Request for creating multiple records with fewer round trips
+	 *
+	 * @param apiVersion Salesforce API version.
+	 * @param allOrNone  Indicates whether to roll back the entire request when the creation of any object fails (true) or to continue with the independent creation of other objects in the request.
+	 * @param records    A list of sObjects.
+	 *
+	 * @see <a href="https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/resources_composite_sobjects_collections_create.htm">https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/resources_composite_sobjects_collections_create.htm</a>
+	 */
+	public static RestRequest getRequestForCollectionCreate(String apiVersion, boolean allOrNone, JSONArray records) throws JSONException {
+		JSONObject requestBodyAsJson = new JSONObject();
+		requestBodyAsJson.put(ALL_OR_NONE, allOrNone);
+		requestBodyAsJson.put(RECORDS, records);
+		return new RestRequest(RestMethod.POST, RestAction.SOBJECT_COLLECTION.getPath(apiVersion), requestBodyAsJson);
+	}
+
+	/**
+	 * Request for retrieving multiple records with fewer round trips
+	 *
+	 * @param apiVersion    Salesforce API version.
+	 * @param objectType    Type of the requested record.
+	 * @param objectIds     List of Salesforce IDs of the requested records.
+	 * @param fieldList     List of requested field names.
+	 *
+	 * @see <a href="https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/resources_composite_sobjects_collections_retrieve.htm">https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/resources_composite_sobjects_collections_retrieve.htm</a>
+	 */
+	public static RestRequest getRequestForCollectionRetrieve(String apiVersion, String objectType, List<String> objectIds, List<String> fieldList)
+		throws UnsupportedEncodingException, JSONException {
+		StringBuilder path = new StringBuilder(RestAction.SOBJECT_COLLECTION_RETRIEVE.getPath(apiVersion, objectType));
+		// Using a post body which is allowed by the end point and allows more ids to be sent up (2000 instead of ~800)
```

Comment:
@bbirman we should use that instead of a "soql where ids in (...)" in briefcase sync down target. It allows for more ids and should be faster.